src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareUtils.java

```diff
@@ -62,4 +62,15 @@ public static HostInfo copyWithAdditionalProps(HostInfo baseHostInfo, Map<String
         mergedProps.putAll(additionalProps);
         return new HostInfo(urlContainer, baseHostInfo.getHost(), baseHostInfo.getPort(), baseHostInfo.getUser(), baseHostInfo.getPassword(), mergedProps);
     }
+
+    public static HostInfo copyWithAdditionalProps(HostInfo baseHostInfo, ConnectionUrl connectionUrl) {
```

Comment:
Can you please add a Javadoc for this method? Helps to distinguish this method from `copyWithAdditionalProps(HostInfo baseHostInfo, Map<String, String> additionalProps)`. It is slightly hard to understand when to use which method.